src/components/Form/Form.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import useInput from '../../hooks/useInput';
+import { Button, Input, TextArea, Rate } from '../';
+import { EMAIL_REGEXP } from '../../config/constants';
+import { Storage } from '../../services/localStorage';
+import * as S from './Form.styles';
+
+const isNotEmpty = value => value.trim() !== '';
+const isEmail = value => value.match(EMAIL_REGEXP);
+const isTextarea = value => value.length;
+
+const BasicForm = () => {
+  const [message, setMessage] = useState('');
+  const [rating, setRating] = useState(0);
+  const {
+    value: userName,
+    isValid: userNameIsValid,
+    hasError: userNameHasError,
+    valueChangeHandler: nameChangeHandler,
+    inputBlurHandler: nameBlurHandler,
+    reset: resetName
+  } = useInput(isNotEmpty);
+
+  const {
+    value: emailValue,
+    isValid: emailIsValid,
+    hasError: emailHasError,
+    valueChangeHandler: emailChangeHandler,
+    inputBlurHandler: emailBlurHandler,
+    reset: resetEmail
+  } = useInput(isEmail);
+
+  const {
+    value: textareaValue,
+    isValid: textareaIsValid,
+    hasError: textareaHasError,
+    valueChangeHandler: textareaChangeHandler,
+    inputBlurHandler: textareaBlurHandler,
+    reset: resetTextarea
+  } = useInput(isTextarea);
+
+  let formIsValid = false;
+
+  if (userNameIsValid && emailIsValid && textareaIsValid) {
+    formIsValid = true;
+  }
+
+  const submitHandler = event => {
+    event.preventDefault();
+
+    if (!formIsValid) {
+      return;
+    }
+
+    const usersFeedback = {
+      id: Math.floor(Math.random() * 10000) * 1,
```

Comment:
why do you need id and what if `Math.random` generates the same number as already stored in local storage?

Reply:
I added dummy id, because usually data has id, I understood that `Math.random` could generate the same number. I will deleted it and leave without dummy ID.